pkg/skaffold/deploy/status_check.go

```diff
@@ -147,13 +177,22 @@ func pollDeploymentStatus(ctx context.Context, runCtx *runcontext.RunContext, r
 	for {
 		select {
 		case <-timeoutContext.Done():
-			msg := fmt.Sprintf("could not stabilize within %v: %v", r.Deadline(), timeoutContext.Err())
-			r.UpdateStatus(proto.ActionableErr{ErrCode: proto.StatusCode_STATUSCHECK_DEADLINE_EXCEEDED,
-				Message: msg})
+			switch c := timeoutContext.Err(); c {
+			case context.Canceled:
+				r.UpdateStatus(proto.ActionableErr{
+					ErrCode: proto.StatusCode_STATUSCHECK_CONTEXT_CANCELLED,
+					Message: "check cancelled\n",
+				})
```

Comment:
Do we really need to log this?  Does it actually happen?   This is single-threaded, so the `cancel()` will only occur when we exit this function.

Reply:
This sounds true, but in actual when a user hits ^C and `cancel()` in invoked, the infinite for-loop does catch run into select cause for `case <-timeoutContext.Done()` and will update the resource status. 
This is good because that way the Skaffold `proto.State` will get updated and `GetState` end point will report the correct statuses.

```
for {
		select {
		case <-timeoutContext.Done():
			switch c := timeoutContext.Err(); c {
			case context.Canceled:
				r.UpdateStatus(proto.ActionableErr{
					ErrCode: proto.StatusCode_STATUSCHECK_CONTEXT_CANCELLED,
					Message: "check cancelled\n",
				})
			case context.DeadlineExceeded:
				r.UpdateStatus(proto.ActionableErr{
					ErrCode: proto.StatusCode_STATUSCHECK_DEADLINE_EXCEEDED,
					Message: fmt.Sprintf("could not stabilize within %v\n", r.Deadline()),
				})
			}
			return
		case <-time.After(pollDuration):
			...
		}
	}
```

Reply:
e.g. on master, run dev and hit Cntrl C as soon as status check starts
```
skaffold dev 

- deployment/nodejs-guestbook-frontend: waiting for rollout to finish: 0 of 1 updated replicas are available...
    - pod/nodejs-guestbook-frontend-5f954f8b9d-h692w: creating container frontend
 - deployment/nodejs-guestbook-mongodb: waiting for rollout to finish: 0 of 1 updated replicas are available...
    - pod/nodejs-guestbook-mongodb-59b8b89f4c-jkmft: creating container mongo
^C - deployment/nodejs-guestbook-backend failed. [2/3 deployment(s) still pending] Error: could not stabilize within 2m0s: context canceled.
 - deployment/nodejs-guestbook-mongodb: could not stabilize within 2m0s: context canceled    - pod/nodejs-guestbook-mongodb-59b8b89f4c-jkmft: creating container mongo
 - deployment/nodejs-guestbook-mongodb failed. [1/3 deployment(s) still pending] Error: could not stabilize within 2m0s: context canceled.
 - deployment/nodejs-guestbook-frontend: could not stabilize within 2m0s: context canceled    - pod/nodejs-guestbook-frontend-5f954f8b9d-h692w: creating container frontend


```

Reply:
IMHO those _context canceled_ items shouldn't be shown.

Reply:
yes.  however, they still appear on master. I will follow up on why they do. It could be because `printStatus` did not have that check which i added in this branch.